src/TimerOutput.jl

```diff
@@ -300,21 +300,50 @@ end
 # Doesn't hurt to keep it for a while though
 timeit(f::Function, label::String) = timeit(f, DEFAULT_TIMER, label)
 function timeit(f::Function, to::TimerOutput, label::String)
-    accumulated_data = push!(to, label)
-    b₀ = gc_bytes()
-    t₀ = time_ns()
+    section = start_timed_section!(to, label)
     local val
     try
         val = f()
     finally
-        accumulated_data.time += time_ns() - t₀
-        accumulated_data.allocs += gc_bytes() - b₀
-        accumulated_data.ncalls += 1
-        pop!(to)
+        stop_timed_section!(to, section)
     end
     return val
 end
 
+struct SectionTimeData
+    label::String # not needed for stopping, but useful for debugging passing this object around
+    data::TimeData
+    allocs_start::Int64
+    time_start::Int64
+end
+
+"""
+    start_timed_section!([to::TimerOutput=DEFAULT_TIMER], label::String)
+
+Start a timed section with the given label. Returns a `SectionTimeData` object that should
+be passed to `stop_timed_section!` when the section is done.
+"""
+start_timed_section!(label::String) = start_timed_section!(DEFAULT_TIMER, label)
+function start_timed_section!(to::TimerOutput, label::String)
+    data = push!(to, label)
+    b₀ = gc_bytes()
+    t₀ = time_ns()
+    return SectionTimeData(label, data, b₀, t₀)
+end
+
+"""
+    stop_timed_section!([to::TimerOutput=DEFAULT_TIMER], section::SectionTimeData)
+
+Stop a timed section started with `start_timed_section!`.
+"""
+stop_timed_section!(section::SectionTimeData) = stop_timed_section!(DEFAULT_TIMER, section)
+function stop_timed_section!(to::TimerOutput, section::SectionTimeData)
+    section.data.time += time_ns() - section.time_start
+    section.data.allocs += gc_bytes() - section.allocs_start
+    section.data.ncalls += 1
+    return pop!(to)
+end
+
```

Comment:
Maybe these should be `begin_timed_section!` and `end_timed_section!` ?

Reply:
`enter_` and `exit_` maybe?

Reply:
Tracy calls them `tracy_emit_zone_begin` so begin/end is probably good.